packages/fast-components-react-base/src/utilities/extractElement.ts

```diff
@@ -0,0 +1,26 @@
+import ReactDOM from "react-dom";
+import { isNil } from "lodash-es";
+
+/**
+ * Attempts to extract an html element from a ref
+ */
+export function extractElementFromRef(
+    sourceRef: React.RefObject<any> | HTMLElement
+): HTMLElement | null {
+    if (sourceRef instanceof HTMLElement) {
+        return sourceRef;
+    }
+    if (!isNil(sourceRef.current)) {
+        if (sourceRef.current instanceof HTMLElement) {
+            return sourceRef.current;
+        }
+
+        const foundNode: Element | Text | null = ReactDOM.findDOMNode(sourceRef.current);
+
+        if (foundNode instanceof HTMLElement) {
+            return foundNode;
+        }
+    }
+
+    return null;
+}
```

Comment:
Could this be useful elsewhere? It seems like since this is somewhat of a generic react utility that `fast-utilities-react` might be a better place for this. Would be great as well to get test coverage.

Reply:
See https://github.com/microsoft/fast-dna/pull/2613
Moved to the utilities in Base components and added tests,